realm-jni/src/io_realm_internal_SharedGroup.cpp

```diff
@@ -83,16 +97,19 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_SharedGroup_nativeCreate(
 }
 
 JNIEXPORT jlong JNICALL Java_io_realm_internal_SharedGroup_createNativeWithImplicitTransactions
-  (JNIEnv* env, jobject, jlong native_replication_ptr, jbyteArray keyArray)
+  (JNIEnv* env, jobject, jlong native_replication_ptr, jint durability, jbyteArray keyArray)
 {
     TR_ENTER()
+
+    SharedGroup::DurabilityLevel level;
+    if (!jintToDurabilityLevel(durability, level)) {
+        ThrowException(env, UnsupportedOperation, "Unsupported durability.");
+        return 0;
+    }
+
     try {
         KeyBuffer key(env, keyArray);
-#ifdef REALM_ENABLE_ENCRYPTION
```

Comment:
`REALM_ENABLE_ENCRYPTION` is an environment variable set by either the builder or Jenkins. It can to have the same value as when core was built. It is true that we for many releases only have had encrypted enabled but I think that by removing it (and unconditionally enable encryption), it will require an extra effort to disable it.